pkg/gds/gds.go

```diff
@@ -254,31 +262,19 @@ func (s *GDS) Register(ctx context.Context, in *api.RegisterRequest) (out *api.R
 				}
 				models.AppendEmailLog(vaspContact, string(admin.ResendVerifyContact), "verify_contact")
 			}
-		} else {
-			// If the model contact is verified make sure that the vasp contact is verified as well
-			if err = models.SetContactVerification(vaspContact, token, true); err != nil {
-				log.Error().Err(err).Str("contact", kind).Str("vasp", vasp.Id).Msg("could not set contact verification token")
-				return nil, status.Error(codes.Aborted, "could not send contact verification emails")
-			}
-			if err = s.db.UpdateVASP(ctx, vasp); err != nil {
-				log.Error().Err(err).Str("vasp", vasp.Id).Msg("could not update contact verification on vasp")
-				return nil, status.Error(codes.Internal, "internal error with registration, please contact admins")
-			}
-
+		} else if vasp.VerificationStatus == pb.VerificationState_SUBMITTED {
+			// If one of the contacts is already verified then short circuit the
+			// verification step and begin the review step by sending a review email to
+			// TRISA admins. This should only be done once for this VASP to avoid
+			// sending the admins duplicate emails.
 			if err = s.beginReview(ctx, vasp, contact.Email); err != nil {
 				log.Error().Err(err).Str("vasp", vasp.Id).Msg("could not begin registration for contact with verified email")
 				return nil, err
 			}
-
-			out = &api.RegisterReply{
-				Id:                  vasp.Id,
-				RegisteredDirectory: vasp.RegisteredDirectory,
-				CommonName:          vasp.CommonName,
-				Status:              vasp.VerificationStatus,
-				Message:             "email successfully verified and verification review sent to TRISA admins",
-			}
-			return out, nil
```

Comment:
This was a bug I think, if we return here then the certificate request doesn't get created. So I changed this to continue through the contact loop but we should only call `beginReview` once per VASP so we don't spam the TRISA admins.